src/connection-manager/dialer/index.ts

```diff
@@ -259,6 +259,8 @@ export class DefaultDialer implements Startable, Dialer {
       async (source) => await all(source)
     )
 
+    addrs = [...new Set(addrs)]
```

Comment:
No concerns for performance and unnecessary allocation of tons of memory for just selecting a list of addresses?

Reply:
Don't think performance or memory will be issue here as we are mostly talking about <10addresses

Reply:
It's not that it's an issue, it that it is unnecessary to tackle this problem this way. This entire function contains multiple un-explicable design choices with the common theme of zero regard for performance in exchange for no clear benefit. A non-piped version would be even easier to read and mantain, and would not require to loop **again** in the Set iteration to drop duplicates

Reply:
As an extreme example: this solution below is correct, and won't affect performance in libp2p overall. But we should never accept it regardless
```suggestion
    addrs = [...new Set([...new Set([...new Set(addrs)])])]
```

Reply:
- Good to see this removed in PR https://github.com/libp2p/js-libp2p/pull/1498/files#diff-f41a0efb6793f2f536fc7b7eee135829b88f2aea74c0391ddb79d382871ca54fL238